### PR TITLE
Changes to enable testing with an IPv6 controlplane

### PR DESCRIPTION
**Pull request by hardys**
Some fixes to allow testing with ipv6 enabled controlplane, the related config looks like:

```
export EXTERNAL_SUBNET="fd2e:6f44:5dd8:c956::/120"
export DNS_VIP="fd2e:6f44:5dd8:c956:0:0:0:2"
export NETWORK_TYPE="OVNKubernetes"
export CLUSTER_SUBNET="fd01::/48"
export CLUSTER_HOST_PREFIX="64"
export SERVICE_SUBNET="fd02::/112"
```

Note this also requires a special release image and some image customizations atm while related OCP/RHCOS fixes land

### CODE REVIEW

File: common.sh [hunk 1]
```diff
@@ -39,13 +39,34 @@ if [ -z "${CONFIG:-}" ]; then
...
+export CLUSTER_NAME=${CLUSTER_NAME:-ostest}
+export CLUSTER_DOMAIN="${CLUSTER_NAME}.${BASE_DOMAIN}"
+export SSH_PUB_KEY="${SSH_PUB_KEY:-$(cat $HOME/.ssh/id_rsa.pub)}"
+export NETWORK_TYPE=${NETWORK_TYPE:-"OpenShiftSDN"}
+export EXTERNAL_SUBNET=${EXTERNAL_SUBNET:-"192.168.111.0/24"}
+export CLUSTER_SUBNET=${CLUSTER_SUBNET:-"10.128.0.0/14"}
+export CLUSTER_HOST_PREFIX=${CLUSTER_HOST_PREFIX:-"23"}
+export SERVICE_SUBNET=${SERVICE_SUBNET:-"172.30.0.0/16"}
+export DNS_VIP=${DNS_VIP:-"192.168.111.2"}
+export LOCAL_REGISTRY_DNS_NAME=${LOCAL_REGISTRY_DNS_NAME:-"virthost.${CLUSTER_NAME}.${BASE_DOMAIN}"}
```
Comment (russellb) [inline common.sh]: want to document the new variables in the readme / config sample?

Comment (hardys) [reply common.sh]: Good point, will do

File: common.sh [hunk 2]
```diff
@@ -39,13 +39,35 @@ if [ -z "${CONFIG:-}" ]; then
 fi
 source $CONFIG
 
+# Provisioning network information
+export PROVISIONING_NETWORK=${PROVISIONING_NETWORK:-172.22.0.0/24}
+export PROVISIONING_NETMASK=${PROVISIONING_NETMASK:-$(ipcalc --netmask $PROVISIONING_NETWORK | cut -d= -f2)}
+export CLUSTER_PRO_IF=${CLUSTER_PRO_IF:-enp1s0}
```
Comment (stbenjam) [inline common.sh]: Duplicated below

File: common.sh (hunk 1 above)
Comment (hardys) [reply common.sh]: Done

File: ocp_install_env.sh [hunk 3]
```diff
@@ -78,12 +69,25 @@ function generate_ocp_install_config() {
     fi
 
     mkdir -p "${outdir}"
+
+    # IPv6 network config validation
+    if [[ "${EXTERNAL_SUBNET}" =~ .*:.* ]]; then
+      if [[ "${NETWORK_TYPE}" != "OVNKubernetes" ]]; then
+        echo "NETWORK_TYPE must be OVNKubernetes when using IPv6"
```
Comment (stbenjam) [inline ocp_install_env.sh]: Could we just set this to OVNKubernetes if we detect IPv6?

**Comment by stbenjam** [comment]
Otherwise LGTM

File: ocp_install_env.sh (hunk 3 above)
Comment (hardys) [reply ocp_install_env.sh]: I guess we could - @russellb do you think anyone is likely to test anything other than OVNKubernetes here?

Comment (russellb) [reply ocp_install_env.sh]: They shouldn't.  I'm OK with just setting it instead of telling them to set it ...

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1456/

File: ocp_install_env.sh (hunk 3 above)
Comment (russellb) [reply ocp_install_env.sh]: let's do this in a follow-up

**Pull request merged**